rlclientlib/live_model_impl.cc

```diff
@@ -496,64 +492,66 @@ int live_model_impl::init_loggers(api_status* status)
   // Get the name of raw data (as opposed to message) sender for interactions.
   const auto* const ranking_sender_impl =
       _configuration.get(name::INTERACTION_SENDER_IMPLEMENTATION, value::get_default_interaction_sender());
-  i_sender* ranking_data_sender = nullptr;
+  std::unique_ptr<i_sender> ranking_data_sender;
 
   // Use the name to create an instance of raw data sender for interactions
   _configuration.set(config_constants::CONFIG_SECTION, config_constants::INTERACTION);
   RETURN_IF_FAIL(_sender_factory->create(
-      &ranking_data_sender, ranking_sender_impl, _configuration, &_error_cb, _trace_logger.get(), status));
+      ranking_data_sender, ranking_sender_impl, _configuration, &_error_cb, _trace_logger.get(), status));
   RETURN_IF_FAIL(ranking_data_sender->init(_configuration, status));
 
   // Create a message sender that will prepend the message with a preamble and send the raw data using the
   // factory created raw data sender
-  l::i_message_sender* ranking_msg_sender = new l::preamble_message_sender(ranking_data_sender);
+  std::unique_ptr<l::i_message_sender> ranking_msg_sender(
```

Comment:
`preamble_message_sender` moves the raw pointer given in its constructor into a unique pointer, so maybe instead of `release()`ing here `preamble_message_sender` constructor could also change to accept a unique pointer

Reply:
same for time provider below (although more interfaces need to change for that)

Reply:
yeah I believe I left comment about it in one place and didn't want to grow PR too much. let's do it in next one